packages/patrol/darwin/Classes/AutomatorServer/Automator/IOSAutomator.swift

```diff
@@ -389,17 +389,33 @@
     func enableDarkMode(_ bundleId: String) throws {
       try runSettingsAction("enabling dark mode", bundleId) {
         #if targetEnvironment(simulator)
-          self.preferences.descendants(matching: .any)["Developer"].firstMatch.tap()
+          // Use system locale directly instead of calling getLocale()
+          let locale = Locale.current.languageCode ?? "en"
+          let localizedStrings = self.getLocalizedStrings(for: locale)
+
+          self.preferences.descendants(matching: .any)[localizedStrings["developer"] ?? "Developer"]
+            .firstMatch.tap()
 
           let value =
-            self.preferences.descendants(matching: .any)["Dark Appearance"].firstMatch.value
+            self.preferences.descendants(matching: .any)[
+              localizedStrings["dark_appearance"] ?? "Dark Appearance"
+            ].firstMatch.value
             as? String?
           if value == "0" {
-            self.preferences.descendants(matching: .any)["Dark Appearance"].firstMatch.tap()
+            self.preferences.descendants(matching: .any)[
+              localizedStrings["dark_appearance"] ?? "Dark Appearance"
+            ].firstMatch.tap()
           }
         #else
-          self.preferences.descendants(matching: .any)["Display & Brightness"].firstMatch.tap()
-          self.preferences.descendants(matching: .any)["Dark"].firstMatch.tap()
+          // Use system locale directly instead of calling getLocale()
+          let locale = Locale.current.languageCode ?? "en"
+          let localizedStrings = self.getLocalizedStrings(for: locale)
+
+          self.preferences.descendants(matching: .any)[
+            localizedStrings["display_brightness"] ?? "Display & Brightness"
```

Comment:
Why do we have this fallback here? We have "en" fallback above already, is this one necessary?

Reply:
This happens in many places